src/modules/aws_codebuild/rpcs/start_build.ts

```diff
@@ -0,0 +1,146 @@
+ * ```
+ *
+ * @see https://github.com/iasql/iasql-engine/blob/main/test/modules/aws-codebuild-integration.ts#L313
+ * @see https://docs.aws.amazon.com/cli/latest/reference/codebuild/start-build.html
+ *
+ */
+export class StartBuildRPC extends RpcBase {
+  /** @internal */
+  module: AwsCodebuildModule;
+
+  /** @internal */
+  outputTable = {
+    name: 'varchar',
+    status: 'varchar',
+    message: 'varchar',
+  } as const;
+
+  /** @internal */
+  startBuild = crudBuilderFormat<CodeBuild, 'startBuild', Build | undefined>(
+    'startBuild',
+    input => input,
+    res => res?.build,
+  );
+
+  /** @internal */
+  call = async (
+    _dbId: string,
+    _dbUser: string,
+    ctx: Context,
+    name: string,
+    region: string,
+  ): Promise<RpcResponseObject<typeof this.outputTable>[]> => {
+    if (!name) {
+      return [
+        {
+          name: '',
+          status: 'ERROR',
+          message: 'Please provide the name of the CodeBuild project to build',
+        },
+      ];
+    }
+
+    // given the project name, read the details
+    const projectObj =
+      (await this.module.project.db.read(
+        ctx,
+        this.module.project.generateId({ projectName: name, region }),
+      )) ??
+      (await this.module.project.cloud.read(
+        ctx,
+        this.module.project.generateId({ projectName: name, region }),
+      ));
+
+    if (projectObj) {
+      const client = (await ctx.getAwsClient(projectObj.region)) as AWS;
+      const cloudBuild = await this.startBuild(client.cbClient, {
+        projectName: name,
+      });
+      if (!cloudBuild || !cloudBuild.id) {
+        return [
+          {
+            name,
+            status: 'KO',
+            message: 'Error launching build',
+          },
+        ];
+      }
+
+      // wait for builds to complete
+      await this.module.buildList.waitForBuildsToComplete(client.cbClient, [cloudBuild.id]);
+
+      // get latest status of the build
+      const awsId = cloudBuild.id;
+
+      const currentBuild = await this.module.buildList.cloud.read(
+        ctx,
+        this.module.buildList.generateId({ awsId, region }),
+      );
+      if (currentBuild) {
+        const dbBuild = await this.module.buildList.buildListMapper(cloudBuild, ctx, projectObj.region);
+        if (!dbBuild) {
+          return [
+            {
+              name,
+              status: 'KO',
+              message: 'Error starting build',
+            },
+          ];
+        } else {
+          // create the builds in the database
+          await this.module.buildList.db.create(dbBuild, ctx);
+
+          return [
+            {
+              name,
+              status: 'OK',
+              message: '',
+            },
+          ];
+        }
+      } else {
+        return [
+          {
+            name,
+            status: 'KO',
+            message: 'Error getting status of build',
+          },
+        ];
+      }
+    } else {
+      return [
```

Comment:
I suggest you add early returns instead of nested conditions. I mean `if (! projectObj) return [...]` and just have the normal behavior in case no returns triggered.